# Parkinson's disease MPRA 

This repository contains scripts used to process sequencing data from Massively Parallel Reporter Assays (MPRA) to determine enhancer activity for candidate elements included within the assay.

# Massively parallel reporter assay (MPRA) data analysis pipeline

This repository contains scripts used to analyse MPRA sequencing 
data
     
## Requirements

* python >3.0
* Biopython
* samtools
* Burrows-Wheeler Aligner (BWA)
* cutadapt
* seqtk

## Usage

::

    python mpra_pipeline.py -h

    usage: mpra_pipeline.py [-h] [-r1 READ1] [-r2 READ2] [-ds1 DNA_SEQ1 [DNA_SEQ1 ...]] [-ds2 DNA_SEQ2 [DNA_SEQ2 ...]] -o OUTDIR [-i INDEX_DIR] -m {enhancer,cdna,gdna}

    MPRA sequencing data analysis pipeline.

    optional arguments:
    -h, --help            show this help message and exit
    -r1 READ1, --read1 READ1
                          Path to a gzipped fastq file. Sequences in this file will be mapped against the reference MPRA library to find refSNP containing reads. Required if run --mode='enhancer'.
    -r2 READ2, --read2 READ2
                          Path to a gzipped fastq file. Sequences in this file will be used to search for barcodes of refSNPs. Required if run --mode='enhancer'.
    -ds1 DNA_SEQ1 [DNA_SEQ1 ...], --dna-seq1 DNA_SEQ1 [DNA_SEQ1 ...]
                          Provide a fastq file or a list of space separated fastq files (forward reads) containing 1) cDNA sequences (required if run --mode='cdna') 2) gDNA sequences (required if run
                          --mode='gdna').
    -ds2 DNA_SEQ2 [DNA_SEQ2 ...], --dna-seq2 DNA_SEQ2 [DNA_SEQ2 ...]
                          Provide a fastq file or a list of space separated fastq files (reverse reads) containing 1) cDNA sequences (required if run --mode='cdna') or 2) gDNA sequences (required if run
                          --mode='gdna').
    -o OUTDIR, --outdir OUTDIR
                          Provide a path to save output files.
    -i INDEX_DIR, --index-dir INDEX_DIR
                          Provide path to the directory where reference index (generated using bwa) files are located and also include prefix of the file
    -m {enhancer,cdna,gdna}, --mode {enhancer,cdna,gdna}
                          Choose a run mode to find the barcodes from enhancer, cDNA or gDNA sequences

## MPRA allele specific and general enhancer analysis 

## RNAseq analysis (CRISPR DEGs)

RNA-seq reads were quality checked using multiQC (v1.13) and mapped against the human reference genome GRCh38 using **STAR (v 2.7.9a)** aligner software using the command:
STAR --runMode alignReads \
         --runThreadN 64 \
         --genomeDir data/GRCh38.p13_indexed_genome \
         --readFilesIn data/sample/sample1_R1.fq.gz data/sample/sample1_R2.fq.gz \
         --readFilesCommand zcat \
         --sjdbGTFfile data/gencode.v43.primary_assembly.annotation.gtf \
         --sjdbOverhang 149 \
         --runRNGseed 444 \
         --outSAMtype BAM SortedByCoordinate \
         --quantMode TranscriptomeSAM \
         --outFileNamePrefix results/star_outputs/sample1 \
        --outTmpDir tmp;

Following read alignment, we used featurecounts module from the **Subread (v2.0.3)** suite to count the number of reads mapping over individual genes. The following command was used to generate a matrix of gene counts per annotated gene in Gencode project (v43) (https://www.gencodegenes.org/). 

featureCounts 
    -a data/gencode.v43.primary_assembly.annotation.gtf \
    -o results/feature_counts/s2/new7122023_count_data.txt \
    results/star_outputs/sample1.sortedByCoord.out.bam \
    -F GTF \
    --largestOverlap \
    -p \
    --countReadPairs \
    -s 2 \
    -B \
    -C \
    -T 64 \
    -t gene \
    -g gene_name \
    --Rpath results/feature_counts/ \
    --tmpDir tmp \
    --verbose

The gene count matrix generated by the featurecounts program was then used as an input for an R package **DESeQ2 (v1.42)** and performed differential gene expression analysis using default settings. We defined differential expressed genes as those with absolute Log2 Fold change difference > 1 and adjusted p-value < 0.01. 
